dependency-manager-tests/carthage/.gitignore

```diff
@@ -1,4 +1,6 @@
 .DS_Store
+/.build
+xcuserdata/
 /Carthage
 
 # Ignore `Cartfile.resolved` and `Cartfile` as they will be re-created for every test run.
```

Comment:
IMO one single `.gitignore` in the parent directory (`/dep-manager-tests`) could simpler and easier to maintain

Reply:
Hmm, maybe let's wait for RUMM-334 where we want to merge `dep-manager-tests` with `instrumented-tests` and we'll see if single `.gitignore` will be practical? IMO it might not, as for Integration and Benchmark we might want to ignore different parts of the setup - at least, each folder has different `.gitignore` now. WDYT?

Reply:
I prefer less `.gitignore` files since they are hidden and not discoverable so i prefer _discover-once-understand-all_

we can have project specific entries as well:
```
...
/carthage/.build
/cocoapods/Pods
...
```

Reply:
KK, will do 👍

Reply:
Done ✅. I also updated SPM test to use `project.pbxproj.src` and gitignored `SPMProject/project.pbxproj`. This way git status is empty after running the test. Now it's also consistent with `Podfile.src` and `Cartfile.src`.